imports/lib/schemas/user.ts

```diff
@@ -12,7 +12,7 @@ const UserCodec = t.type({
   createdAt: date,
   lastLogin: t.union([date, t.undefined]),
   services: t.union([t.object, t.undefined]),
-  roles: t.union([t.array(t.string), t.undefined]),
+  roles: t.union([t.object, t.undefined]),
```

Comment:
Hmmm, does io-ts give us the ability to express that this object is a `Record<string, string[]>`?

Reply:
io-ts does, but I don't think SimplSchema does. As far as I can tell, it can only represent an object with a fixed set of keys.

We don't actually use the io-ts generated codec anywhere, because there's no way to attach it to the `Meteor.users` collection that already exists. Instead, we add our modifications to `Meteor.users` with an [ambient type declaration](https://github.com/deathandmayhem/jolly-roger/blob/2f7d6c973f67855afc1f60fb24bf51f7643c9c0d/types/user.d.ts). Honestly, we should probably not use io-ts here at all, and instead just create a SimplSchema directly.